cmake/BipedalLocomotionFrameworkDependencies.cmake

```diff
@@ -3,62 +3,82 @@
 # GNU Lesser General Public License v2.1 or any later version.
 
 include(BipedalLocomotionFrameworkFindDependencies)
+include(BipedalLocomotionDependencyClassifier)
 
 ################################################################################
-# Find all packages
+########################## Mandatory dependencies ##############################
+
+find_package(iDynTree 3.0.0 REQUIRED)
+checkandset_dependency(iDynTree)
```

Comment:
This would allow the user to disable the use of ``iDyntree``, that is unwanted. Maybe, instead of having 
```
  if(${PREFIX}_USE_${package})
```
you could use another input of ``depedency_classifier`` and move that variable outside. For example, this additional input might be called ``IS_USED``, and in case of strong dependencies like ``iDyntree`` you would have
```
dependency_classifier(iDynTree 
                                     MINIMUM_VERSION 3.0.0
                                     PUBLIC
                                     IS_USED TRUE)
```
while for ``yarp`` it would be
```
dependency_classifier(YARP 
                                     PUBLIC
                                     IS_USED ${FRAMEWORK_USE_YARP})
```

Reply:
Done here: https://github.com/dic-iit/bipedal-locomotion-framework/pull/252/commits/d2ba1d5b35fa7fba20e54996df6fa699d6265210